### DISCUSSION TITLE
cartLinesUpdate removes properties that are not being passed.

### DISCUSSION THREAD

**Discussion by ogaguinaga** [Shopify/storefront-api-feedback#170] [Ideas & Requests]
There is currently an issue using `CartLineQuantityAdjustButton` when trying to increase or decrease the quantity of a specific `cartLineItem` with an existing `attributes` property. Currently it increases and decreases the quantity correctly but removes all other properties that were originally on the `cartLine`. [Here is a reproduction of this issue](https://stackblitz.com/edit/shopify-hydrogen-dahx9u?file=src/components/product/ProductForm.client.jsx). In order to replicate follow the following steeps.

1.  Add any product to the cart. I harcoded an attribute `attributes={[{key: 'Engraving', value: 'Hello world'}]}`
2. Open cart, on the console it will log the cart item
3. Increase the number of items (for me, in my app, doing this removes the attributes, but for some reason on this example it does no, but...)
4. Increase by one more so now you have 3 of the same product, now it removed the attributes from the cart item.

After discussing this issue on the Discord channel with someone from the Shopify Hydrogen Team he was able to figure out the following:

1. The `CartLineQuantityAdjustButton` calls `linesUpdate()` here https://github.com/Shopify/hydrogen/blob/767a4ac15c112324ab885863600e9329e1b1416d/packages/hydrogen/src/components/CartLineQuantityAdjustButton/CartLineQuantityAdjustButton.tsx#L36 and only passes in the ID and Quantity
2. That in turn eventually calls the Storefront API's `CartLineUpdate()` method here https://github.com/Shopify/hydrogen/blob/767a4ac15c112324ab885863600e9329e1b1416d/packages/hydrogen/src/components/CartProvider/CartActions.client.tsx#L122 and here https://github.com/Shopify/hydrogen/blob/767a4ac15c112324ab885863600e9329e1b1416d/packages/hydrogen/src/components/CartProvider/cart-queries.ts#L39
3. It appears the Storefront API's `cartLinesUpdate` mutation does not keep existing properties (such as attributes) if you don't pass them into the mutation. In this case, the component is only passing in the ID and Quantity, and not the Attributes, so those get lost as the quantity is updated.

And as a solution he suggested to update the `cartLinesUpdate()` so that when you call it, it only updates what you pass to it and doesn't remove/delete things that you don't pass to it. Looking forward to hearing you opinion on this.

**Comment by seandogg** [comment]
Was struggling for a long time trying to get this working, grateful I found this post to understand what's going on under the hood. Does anyone have a work around for it?

**Comment by vixdug** [comment]
Hi @ogaguinaga - Thanks for bringing this to our attention. `cartLinesUpdate` should not remove existing attributes if they are not passed into the update mutation. The team is aware and I will update here once this fix ships.

**Comment by clarkey** [comment]
Came across this issue today. Can we revisit this idea? @vixdug
